### PR TITLE
C API instantiate with resolving imported functions

### PR DESCRIPTION
**Pull request by gumb0**
Replaces https://github.com/wasmx/fizzy/pull/565

### CODE REVIEW

File: include/fizzy/fizzy.h [hunk 1]
```diff
@@ -130,6 +141,22 @@ bool fizzy_find_exported_function(
...
+///
+/// @param      module                   Pointer to module.
+/// @param      imported_functions       Pointer to the imported function array. Can be NULL iff
+///                                      imported_functions_size equals 0.
+/// @param      imported_functions_size  Size of the imported function array. Can be zero.
+/// @returns    non-NULL pointer to instance in case of success, NULL otherwise.
+///
+/// @note
+/// Functions in @a imported_functions are allowed to be in any order and allowed to include some
+/// functions not required by instantiated module.
```
Comment (axic) [inline include/fizzy/fizzy.h:155]:
```suggestion
/// functions not required by the module.
```

File: test/unittests/capi_test.cpp [hunk 2]
```diff
@@ -190,7 +283,8 @@ TEST(capi, memory_access)
       )
     */
     const auto wasm = from_hex(
-        "0061736d010000000105016000017f0302010005030100010a0901070041002802000b0b08010041010b02112"
+        "0061736d010000000105016000017f0302010005030100010a0901070041002802000b0b08010041010b02"
+        "112"
         "2");
```
Comment (axic) [inline test/unittests/capi_test.cpp:288]: Why are these changing?

Comment (gumb0) [reply test/unittests/capi_test.cpp]: I don't know why it did, reverted.

**Review by axic — APPROVED** [review]
Looks okay to me. @chfast ?

**Review by chfast — APPROVED** [review]

**Pull request merged**